32328570
Elasticsearch Two sibling aggregations cannot have the same name
<p>I'm running Elastic 1.4.4 and have created some code with a range aggregation and a sub-aggregation (cardinality).&#xA;If I run my code with a matchall query and some filterquery to get a subset all works fine.</p>&#xA;&#xA;<p>I see the range aggregation and the expected values for the sub-aggregation.</p>&#xA;&#xA;<p>But as soon as I add a query things fall apart.&#xA;I run into an error: "Two sibling aggregations cannot have the same name".</p>&#xA;&#xA;<p>This seems strange since I have the same aggregation/sub-aggregation defined before without any problem.</p>&#xA;&#xA;<p>This is basically what I do:</p>&#xA;&#xA;<pre><code>SearchResponse response = &#xA;client.prepareSearch(esIndex)&#xA;.setQuery(query)&#xA;.setFrom(startAt)&#xA;.setSize(topSize)&#xA;.addAggregation(AggregationBuilders.cardinality(UNIQUE_IPS).field(Constants.MAIN_COLLAPSE_FIELD))&#xA;.addAggregation(collapse)&#xA;.addAggregation(filtersetbranche)              \&#xA;.addAggregation(Constants.FTE_RANGES_AGGS.subAggregation(AggregationBuilders.cardinality("ftecounts").field(Constants.MAIN_COLLAPSE_FIELD)))&#xA;.execute()&#xA;.actionGet();&#xA;</code></pre>&#xA;
<p>I found the issue.</p>&#xA;&#xA;<p>I added a subaggregation to the Constants.FTE_RANGES_AGGS dynamically.&#xA;So, the first time I call this piece of code all is fine.&#xA;But the next time I added the same subaggregation with the same name hence the error "Two sibling aggregations cannot have the same name".</p>&#xA;&#xA;<p>So, I have to rewrite it to use a copy of the Constant or do it in a different way.</p>&#xA;